docs/final report/chapters/engineering-reflection.tex

```diff
@@ -1,12 +1,32 @@
 \chapter{Engineering Reflection} %MAX 1 PAGE
 
-Introduce the chapter, explain what will be descibed. 2 PAGES
+In this section we will describe how the engineering process went. We discuss how we have used scrum to develop the product. Furthermore we will discuss the GitHub pull requests and how we ensured that our code had good quality. As last we will discuss the feedback we got from SIG.
+
+\section{Scrum}
+We used the scrum work flow to develop the product. On Friday we created a plan for the coming week. Every week-day we started with a short daily-meeting. In that meeting we discussed what we had done, what we were going to do and which problems we encountered. At Friday our sprint ended. We created a sprint-reflection and we gave a demo to the client. Next we started a new sprint. 
+At the start of the project we underestimated the time needed for tasks. Therefore a lot of task were not done during a sprint. Later in the project we estimated it better and therefore there were less incompleted tasks.
+
+We liked that scrum was focused on features. But we also noticed that because the focused on features there is no room for task that can not be translated into a feature. While those task could be important. 
 
 \section{GitHub Pulls}
-Talk about the way we handled GitHub pulls
+It was not allowed to push changes directly to the master. When a person wanted to merge a change into the master he had to open a pull request. The pull request had to be approved by at least two team members before it could be merged with the master. The code for the pull request should have good quality (see section \ref{codequality}).  We looked very critical to pull request. So often the one who has requested the pull request had to improve some parts of his code. This led to good code quality. However most comments were related to code style and not directly to the implementation of the code.
+
+The size of the pull requests varies a lot. In the beginning of the project they were very large. But this made them hard to review and caused them to stay open for a long time. Therefore we tried to create smaller pull requests. This result of this was that pull requests were handled quicker and that we could review them better. However there were still some very large pull requests.
 
-\section{Code Quality}
-Talk about the way we maintained quality in the code
+Reviewing the code resulted in good code quality. Therefore we will try use the pull request in future projects. However we will then try to keep the pull requests small. 
```

Comment:
pull based development ipv. the pull request. However we will try vervangen door We'll remember to keep